android/src/main/java/io/ably/lib/rest/AblyRest.java

```diff
@@ -36,6 +38,24 @@ public AblyRest(ClientOptions options) throws AblyException {
         super(options, new AndroidPlatformAgentProvider());
     }
 
+    /**
+     * Constructor implementation to be able to have shallow copy of the client,
+     * allowing us to modify certain fields while implementing a proxy for the Realtime/Rest SDK wrapper
+     */
+    protected AblyRest(AblyRest underlyingClient, HttpCore httpCore, Http http) {
+        super(underlyingClient, httpCore, http);
+    }
+
+    /**
+     * [Internal Method]
+     * <p/>
+     * We use this method to create a shallow copy of the client, allowing us to modify certain fields
+     * while implementing a proxy for the Realtime/Rest SDK wrapper
+     */
+    public AblyRest createShallowCopy(HttpCore httpCore, Http http) {
```

Comment:
We only have `createWrapperSDKProxy` as a public method right? maybe we should mark this as protected.
If it's not accessible, we can access it using [invoke private java method](https://github.com/ably/ably-chat-kotlin/blob/71b20c4b70c93fd91d3356d743843da514df70cb/chat-android/src/test/java/com/ably/chat/TestUtils.kt#L110) ( just need to remove suspendCancellableCoroutine )

Reply:
This is to make sure we don't create out of spec public methods

Reply:
Good spot, I guess we can make it protected, I think we don't use it outside the package. Let me check. But "invoke private java method" relies on reflection and tampering with method visibility, which is always best to avoid

Reply:
refactored a bit, get rid of this code

Reply:
Great